src/core/qgsstoredexpressionmanager.cpp

```diff
@@ -0,0 +1,180 @@
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsstoredexpressionmanager.h"
+
+#include <QDomElement>
+
+QUuid QgsStoredExpressionManager::addStoredExpression( const QString &name, const QString &expression, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  /* leads to trouble with sip...
+  QgsStoredExpression storedExpression( name, expression );
+  */
+  QgsStoredExpression storedExpression;
+  storedExpression.id = QUuid::createUuid();
+  storedExpression.name = name;
+  storedExpression.expression = expression;
+
+  mStoredExpressions.append( storedExpression );
+
+  return storedExpression.id;
+}
+
+void QgsStoredExpressionManager::removeStoredExpression( const QUuid &id, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  int i = 0;
+  for ( const QgsStoredExpression &storedExpression : mStoredExpressions )
+  {
+    if ( storedExpression.id == id )
+    {
+      mStoredExpressions.removeAt( i );
+      break;
+    }
+    ++i;
+  }
+}
+
+void QgsStoredExpressionManager::updateStoredExpression( const QUuid &id, const QString &name, const QString &expression, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  int i = 0;
+  for ( const QgsStoredExpression &storedExpression : mStoredExpressions )
+  {
+    if ( storedExpression.id == id )
+    {
+      QgsStoredExpression newStoredExpression = mStoredExpressions.at( i );
+      newStoredExpression.name = name;
+      newStoredExpression.expression = expression;
+      mStoredExpressions.replace( i, newStoredExpression );
+      break;
+    }
+    ++i;
+  }
+}
+
+void QgsStoredExpressionManager::addStoredExpressions( QList< QgsStoredExpression > storedExpressions, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  mStoredExpressions.append( storedExpressions );
+}
+
+QList< QgsStoredExpression > QgsStoredExpressionManager::storedExpressions( const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  return mStoredExpressions;
+}
+
+QgsStoredExpression QgsStoredExpressionManager::storedExpression( const QUuid &id, const QString &tag )
+{
+  Q_UNUSED( tag );
+
+  for ( const QgsStoredExpression &storedExpression : mStoredExpressions )
+  {
+    if ( storedExpression.id == id )
+    {
+      return storedExpression;
+    }
+  }
+  return QgsStoredExpression();
+}
+
+QgsStoredExpression QgsStoredExpressionManager::findStoredExpressionByExpression( const QString &expression, const QString &tag )
```

Comment:
```suggestion
QgsStoredExpression QgsStoredExpressionManager::findStoredExpressionByExpression( const QString &expression, const QString &tag ) const
```

This shouldn't modify the internals?